<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ServiceConfigurationPage</class>
 <widget class="QWidget" name="ServiceConfigurationPage">
  <property name="windowTitle">
   <string notr="true">Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>General</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_12">
      <item row="0" column="0" colspan="5">
       <widget class="QCheckBox" name="isTrayIconHidden">
        <property name="text">
         <string>Hide tray icon</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="5">
       <widget class="QCheckBox" name="autostartService">
        <property name="text">
         <string>Autostart</string>
        </property>
       </widget>
      </item>
      <item row="6" column="4">
       <widget class="QPushButton" name="stopService">
        <property name="text">
         <string>Stop service</string>
        </property>
        <property name="icon">
         <iconset resource="../veyon-configurator.qrc">
          <normaloff>:/resources/media-playback-stop.png</normaloff>:/resources/media-playback-stop.png</iconset>
        </property>
       </widget>
      </item>
      <item row="6" column="2">
       <spacer name="horizontalSpacer_9">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="6" column="1">
       <widget class="QLabel" name="serviceState">
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Stopped</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="5">
       <widget class="QCheckBox" name="isSoftwareSASEnabled">
        <property name="text">
         <string>Enable SAS generation by software (Ctrl+Alt+Del)</string>
        </property>
       </widget>
      </item>
      <item row="6" column="3">
       <widget class="QPushButton" name="startService">
        <property name="text">
         <string>Start service</string>
        </property>
        <property name="icon">
         <iconset resource="../veyon-configurator.qrc">
          <normaloff>:/resources/media-playback-start.png</normaloff>:/resources/media-playback-start.png</iconset>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="5">
       <widget class="QCheckBox" name="isMultiSessionServiceEnabled">
        <property name="toolTip">
         <string>Enabling this option will make the service launch a server process for every interactive session on a computer.
Typically this is required to support terminal servers.</string>
        </property>
        <property name="text">
         <string>Multi session support (experimental)</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="5">
       <widget class="QCheckBox" name="remoteConnectionNotificationsEnabled">
        <property name="text">
         <string>Show notification on remote connection</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="label_16">
        <property name="text">
         <string>State:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="5">
       <widget class="QCheckBox" name="failedAuthenticationNotificationsEnabled">
        <property name="text">
         <string>Show notification on failed authentication attempts</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_8">
     <property name="title">
      <string>Network</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_11">
      <item row="2" column="1">
       <widget class="QSpinBox" name="featureWorkerManagerPort">
        <property name="minimum">
         <number>1024</number>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>Demo server port</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="primaryServicePort">
        <property name="minimum">
         <number>1024</number>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_14">
        <property name="text">
         <string>Feature manager port</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>Primary service port</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QCheckBox" name="isFirewallExceptionEnabled">
        <property name="text">
         <string>Enable firewall exception</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="demoServerPort">
        <property name="minimum">
         <number>1024</number>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="2">
       <widget class="QCheckBox" name="localConnectOnly">
        <property name="text">
         <string>Allow connections from localhost only</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_15">
        <property name="text">
         <string>Internal VNC server port</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="vncServerPort">
        <property name="minimum">
         <number>1024</number>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="vncServerGroupBox">
     <property name="title">
      <string>VNC server</string>
     </property>
     <layout class="QVBoxLayout" name="vncServerGroupBoxLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Plugin:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="vncServerPlugin">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>isTrayIconHidden</tabstop>
  <tabstop>failedAuthenticationNotificationsEnabled</tabstop>
  <tabstop>remoteConnectionNotificationsEnabled</tabstop>
  <tabstop>isSoftwareSASEnabled</tabstop>
  <tabstop>isMultiSessionServiceEnabled</tabstop>
  <tabstop>autostartService</tabstop>
  <tabstop>startService</tabstop>
  <tabstop>stopService</tabstop>
  <tabstop>primaryServicePort</tabstop>
  <tabstop>vncServerPort</tabstop>
  <tabstop>featureWorkerManagerPort</tabstop>
  <tabstop>demoServerPort</tabstop>
  <tabstop>isFirewallExceptionEnabled</tabstop>
  <tabstop>localConnectOnly</tabstop>
  <tabstop>vncServerPlugin</tabstop>
 </tabstops>
 <resources>
  <include location="../veyon-configurator.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>vncServerPlugin</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>ServiceConfigurationPage</receiver>
   <slot>updateVncServerPluginConfigurationWidget()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>381</x>
     <y>703</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>484</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startService</sender>
   <signal>clicked()</signal>
   <receiver>ServiceConfigurationPage</receiver>
   <slot>startService()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>230</x>
     <y>118</y>
    </hint>
    <hint type="destinationlabel">
     <x>207</x>
     <y>234</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stopService</sender>
   <signal>clicked()</signal>
   <receiver>ServiceConfigurationPage</receiver>
   <slot>stopService()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>342</x>
     <y>118</y>
    </hint>
    <hint type="destinationlabel">
     <x>207</x>
     <y>234</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updateVncServerPluginConfigurationWidget()</slot>
  <slot>startService()</slot>
  <slot>stopService()</slot>
 </slots>
</ui>
